### PR TITLE
Combo box strict typing

### PR DESCRIPTION
**Pull request by JoHeyJo**
start: 28 errors
end: 10 errors

### CODE REVIEW

**Comment by JoHeyJo** [comment]
Screen Shot 2023-04-20 at 11 36 02 AM

this is the error I was referring to. Can I believe to correct this I need to make that property optional by just adding a `?`  But I remember you saying something about not manually updating some file... Was it the supabase.ts file?

File: components/Combobox.tsx [hunk 1]
```diff
@@ -2,15 +2,28 @@ import classNames from '../modules/classnames';
...
+  onSelect: (category: categoryApi.Category) => void;
+  createOption: (categoryName: string) => Promise<categoryApi.Category | undefined>;
+}
+
+type Option = {
+  name: string;
+}
+
+
+export default function Dropdown({ options, onSelect, createOption, preselectedValue = { created_at: '', id: 0, name: '', user_id:'' } }: DropdownProps) {
```
Comment (JoHeyJo) [inline components/Combobox.tsx:19]: I'm a little unsure of how I resolved this type error. Is this the best way to go about it?

**Comment by lananelson** [comment]
soo this is a bit tricky from data base perspective id should always be defined we cannot make it optional. So this is one of those situations where you might need to refactor some stuff to allow both an option for "+create" without an id and real options with ids.

**Comment by lananelson** [comment]
also related: we should never change types in the generated supabase files manually, the idea is that they always reflect the types in database and they should not be out of sync with the database. Eventually we need our tools to run automatically on every commit to re-generate the types (right now it is a manual process)

**Comment by lananelson** [comment]
Good job! Shipppppit

**Pull request merged**